src/bookmarklet/app.js

```diff
@@ -60,20 +58,25 @@
             // load config from inner CDN
             configOrigin = `https://${ref}--${repo}--${owner}.hlx.live`;
           }
-          const configScript = document.createElement('script');
-          configScript.id = 'hlx-sk-config';
-          configScript.src = `${configOrigin}/tools/sidekick/config.js`;
-          configScript.addEventListener('error', () => {
-            // if no project config, init sidekick with base config
-            console.info(`no sidekick config found at ${configScript.src}`);
-            window.hlx.initSidekick();
-          });
-          // init sidekick via project config
-          if (document.getElementById(configScript.id)) {
-            document.getElementById(configScript.id).replaceWith(configScript);
-          } else {
-            document.head.append(configScript);
+          try {
+            const res = await fetch(`${configOrigin}/tools/sidekick/config.json`);
+            if (res.ok) {
+              console.log('custom sidekick config found');
+              // apply custom config
+              window.hlx.sidekickConfig = {
+                ...window.hlx.sidekickConfig,
+                ...(await res.json()),
+              };
```

Comment:
```suggestion
              Object.apply(window.hlx.sidekickConfig, await res.json());
```

Reply:
That's what I had before, but I thought the spread operator notation should be preferred? `eslint` didn't seem to have an opinion either way though...